miden/src/repl/mod.rs

```diff
@@ -270,18 +269,18 @@ fn execute(program: String) -> Result<(Vec<(u64, Word)>, Vec<Felt>), ProgramErro
     let stack_inputs = StackInputs::default();
     let advice_provider = MemAdviceProvider::default();
 
-    let mut process = Process::new_debug(program.kernel().clone(), stack_inputs, advice_provider);
-    let _program_outputs = process.execute(&program).map_err(ProgramError::ExecutionError);
-
-    let (sys, _, stack, _, chiplets, _) = process.into_parts();
+    let state_iter = processor::execute_iter(&program, stack_inputs, advice_provider);
+    let (system, _, stack, chiplets, err) = state_iter.into_parts();
+    if let Some(err) = err {
+        Err(err)
```

Comment:
I believe this should be `return Err(err);`.

Reply:
Exactly, fixed in 642e6bcc

Reply:
Ah - maybe `return err.map_err(ProgramError::ExecutionError);`?

Reply:
Oh finally in 373bb68f 😆